dotnet/src/Connectors/VectorData.Abstractions/RecordOptions/FilteredRecordRetrievalOptions.cs

```diff
@@ -39,76 +38,10 @@ public int Skip
     /// <value>
     /// If not provided, the order of returned results is non-deterministic.
     /// </value>
-    public OrderByDefinition OrderBy { get; } = new();
+    public Func<OrderByDefinition<TRecord>, OrderByDefinition<TRecord>>? OrderBy { get; set; }
 
     /// <summary>
     /// Gets or sets a value indicating whether to include vectors in the retrieval result.
     /// </summary>
     public bool IncludeVectors { get; set; }
-
-    /// <summary>
-    /// A builder for sorting.
-    /// </summary>
-    // This type does not derive any collection in order to avoid Intellisense suggesting LINQ methods.
-    public sealed class OrderByDefinition
```

Comment:
I've moved it out of the option bag as IIRC that was the recommendation during API review

Reply:
I don't remember that part...

I slightly prefer it inside, since this isn't actually a type that users should access directly, and if it's outside it'll show up in Intellisense etc... Otherwise what do we gain by moving it out?

Reply:
> I don't remember that part...

https://www.youtube.com/watch?v=IYvmkh8_7dw&t=4617s

Jeremy said something between the lines of "In general, nested types are discouraged because they are hard to work with." Later he added "nobody is using new so it should be fine". But with my `Func` change, the authors of the connectors may need to use `new` (however `new()` will also work).

 > Otherwise what do we gain by moving it out?

The ability to reuse it the future for anything that would like to define the ordering of records.

Reply:
Yeah, IMHO this change makes it slightly nicer for connector implementors (we don't really care about much that side), at the expense of introducing potentially more Intellisense/suggestions confusion on the user side (which we do care about).

So I'd prefer if we kept this nested...